src/antares/service/api_services/area_api.py

```diff
@@ -479,71 +483,75 @@ def update_area_ui(self, area: Area, ui: AreaUi) -> AreaUi:
 
             url = f"{base_url}?type=AREA&ui=true"
             response = self._wrapper.get(url)
-            json_ui = response.json()[area.id]
+            json_ui = response.json()[area_id]
             ui_response = AreaUiResponse.model_validate(json_ui)
             area_ui = AreaUi.model_validate(ui_response.to_craft())
 
         except APIError as e:
-            raise AreaUiUpdateError(area.id, e.message) from e
+            raise AreaUiUpdateError(area_id, e.message) from e
 
         return area_ui
 
     def delete_area(self, area: Area) -> None:
```

Comment:
The delete area should only get the area_id to be the same as the other methods. The delete inisde the sutdy.py gets an Area but calls the service with the id